core/block.go

```diff
@@ -92,57 +92,59 @@ func networkBlockHashMetaInfo(network utils.Network) *blockHashMetaInfo {
 	}
 }
 
+type BlockCommitments struct {
+	TransactionCommitment *felt.Felt
+	EventCommitment       *felt.Felt
+}
+
 // VerifyBlockHash verifies the block hash. Due to bugs in Starknet alpha, not all blocks have
 // verifiable hashes.
-func VerifyBlockHash(b *Block, network utils.Network) error {
+func VerifyBlockHash(b *Block, network utils.Network) (*BlockCommitments, error) {
 	if len(b.Transactions) != len(b.Receipts) {
-		return fmt.Errorf("len of transactions: %v do not match len of receipts: %v",
+		return nil, fmt.Errorf("len of transactions: %v do not match len of receipts: %v",
 			len(b.Transactions), len(b.Receipts))
 	}
 
 	for i, tx := range b.Transactions {
 		if !tx.Hash().Equal(b.Receipts[i].TransactionHash) {
-			return fmt.Errorf(
+			return nil, fmt.Errorf(
 				"transaction hash (%v) at index: %v does not match receipt's hash (%v)",
 				tx.Hash().String(), i, b.Receipts[i].TransactionHash)
 		}
 	}
 
 	if err := VerifyTransactions(b.Transactions, network, b.ProtocolVersion); err != nil {
-		return err
+		return nil, err
 	}
 
 	metaInfo := networkBlockHashMetaInfo(network)
-
 	unverifiableRange := metaInfo.UnverifiableRange
-	if unverifiableRange != nil {
-		// Check if the block number is in the unverifiable range
```

Comment:
Can we calculate the commitments here so we don't do the check in the loop?

Reply:
I considered that too, but didnt want to duplicate the `blockHash(b, network, overrideSeq)` call.

Reply:
Can we export the `transactionCommitment` and `eventCommitment` functions, call them from `SanityCheckNewHeight` (or wherever), and pass them down to `blockHash` instead of bubbling them up? Injects the dependencies.

Reply:
They are not calculated the same for all block versions and some blocks dont even use `eventCommitment`. Additional logic in `blockHash`, `pre07Hash` and `post07Hash` handles these differences. So we would need to move (if not duplicate, which is worse) those checks out of `blockHash` and into the caller site. 

I want to keep the complexity close to wherever the data is actually used, so I think it is best to keep it as is for now.

Reply:
Sounds good.